### ISSUE TITLE
Remove-Migration for last migration fails to remove model snapshot (at least on UWP apps)

### ISSUE THREAD

**Issue by divega** [aspnet/EntityFramework#5293]
### Repro steps:
1. Create UWP application as described in http://docs.efproject.net/en/latest/platforms/uwp/getting-started.html (I haven't tried with other kinds of applications)
2. Stop after add-migrations step succeeds 
3. Execute `Remove-Migration -Verbose`

### Expected result:
Both last (and only) migration and model snapshot are removed.

### Actual result: 
Output is:
```
Using startup project 'App2'.
Using project 'App2'
Build started...
Build succeeded.
Using application base 'C:\Users\?\documents\visual studio 2015\Projects\App2\App2\bin\x86\Debug\'.
No configuration file found.
Using current directory 'C:\Users\?\documents\visual studio 2015\Projects\App2\App2\bin\x86\Debug\'.
Finding DbContext classes...
Using context 'BloggingContext'.
Removing migration '20160507012207_MyFirstMigration' without checking the database. If this migration has been applied to the database, you will need to manually reverse the changes it made.
Removing migration '20160507012207_MyFirstMigration'.
Removing model snapshot.
Done.
Value does not fall within the expected range.
```
(notice the error message at the end)

Since the snapshot still exists after this, attempting to add the migration again will create an empty migration

**Comment by divega** [comment]
I having problems to reproduce this with the latest build, i.e. the snapshot is now removed correctly. I am unfortunately sure what was causing the issue before 😞  but leaving the issue open just in case the error "Value does not fall within the expected range." is a helpful clue.

**Comment by natemcmaster** [comment]
The error  "Value does not fall within the expected range" is the same error thrown by #5383 which is fixed by #5476.
